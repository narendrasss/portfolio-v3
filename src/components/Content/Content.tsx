import React from 'react';
import Helmet from 'react-helmet';
import Header from '../Header/Header';
import Main, { MainProps } from '../Main/Main';
import NavButton from '../NavButton/NavButton';

export interface ContentProps extends MainProps, SiteMetadata {
  title: string;
  onNavClick: React.ChangeEventHandler<HTMLButtonElement>;
}

const Content: React.FC<ContentProps> = ({
  title,
  description,
  url,
  links,
  children,
  onNavClick,
  ...props
}) => (
  <Main data-testid="main" p={1} {...props}>
    <Helmet
      title={title}
      meta={[{ name: 'description', content: description }]}
    />
    <Header url={url} links={links} />
    <NavButton
      data-testid="nav-button"
      onClick={onNavClick}
      display={['block', 'none']}
    />
    {children}
  </Main>
);

export default Content;
