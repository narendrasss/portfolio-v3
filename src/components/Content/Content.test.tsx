import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import Content from './Content';

describe('Content', () => {
  const props = {
    title: `Narendra Syahrasyad`,
    description: `Front-end web developer and Business and Computer Science candidate at UBC.`,
    url: `narendras.io`,
    links: [
      {
        name: `projects`,
        url: `/projects`,
      },
      {
        name: `about`,
        url: `/about`,
      },
      {
        name: `blog`,
        url: `/blog`,
      },
      {
        name: `resume`,
        url: `/resume`,
      },
    ],
    onNavClick: jest.fn(),
  };

  it('calls handler on nav button press', () => {
    const { getByTestId } = render(<Content {...props} />);

    fireEvent.click(getByTestId('nav-button'));
    expect(props.onNavClick).toHaveBeenCalled();
  });
});
