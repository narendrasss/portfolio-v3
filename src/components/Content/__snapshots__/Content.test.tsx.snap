// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders without crashing 1`] = `
<DocumentFragment>
  <main
    class="sc-bdVaJa sc-bwzfXH sc-gZMcBi gHDtBQ"
    data-testid="main"
  >
    <header
      class="sc-bdVaJa sc-bwzfXH sc-iwsKbI gDIeTP"
    >
      <a
        class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat cYLLdT"
        color="black"
        font-family="sans"
        font-size="0,1"
        font-weight="600"
        href="/home"
      >
        narendras
        <span
          class="sc-bdVaJa sc-htpNat fYauba"
          font-family="sans"
          font-size="0,1"
          font-weight="300"
        >
          .io
        </span>
      </a>
      <nav
        class="sc-bdVaJa sc-dnqmqq hjWSCW"
        display="none,block"
      >
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/projects"
        >
          projects
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/about"
        >
          about
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/blog"
        >
          blog
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/resume"
        >
          resume
        </a>
      </nav>
    </header>
    <button
      class="sc-bdVaJa sc-gqjmRU sc-VigVT bvdyGS sc-dnqmqq kXPwVz"
      data-testid="nav-button"
      display="block,none"
      type="button"
      width="4.5rem"
    >
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX hFTANy"
        width="4.5rem"
      />
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX cVrGbe"
        width="3.6rem"
      />
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX gTeBaA"
        width="2.25rem"
      />
    </button>
  </main>
</DocumentFragment>
`;
